client.go

```diff
@@ -857,6 +857,20 @@ type link struct {
 	msg           Message       // current message being decoded
 }
 
+func (l *link) getSource() *source {
+	if l.source == nil {
+		l.source = new(source)
+	}
+	return l.source
+}
+
+func (l *link) getTarget() *target {
+	if l.target == nil {
+		l.target = new(target)
+	}
+	return l.target
+}
```

Comment:
I'm ok with init methods, but I'd like to avoid getters returning pointers that are then modified.

Reply:
These aren't initers, this is constructing a blank object demand. There are many places where you want to set a target/source field, but first have to make sure the target/source exists. I think a method for that is better than repeating the if target == nil {} logic everywhere - that's how the SEGV was caused - in one place someone forgot it.

Reply:
As I stated, I disagree and won't be merging this.